crates/cli-tools/src/changelog.rs

```diff
@@ -872,10 +914,146 @@ mod tests {
         let mut changelog =
             Changelog::parse("", changelog_str).expect("Failed to parse changelog.");
 
-        let current_release = changelog.get_or_create_release_mut();
+        changelog.push_description(Severity::Patch, "asdf").expect("Failed to push description.");
+
+        assert_eq!(changelog.get_current_release().version, Version::parse("0.2.1-git").unwrap());
+    }
+
+    #[test]
+    fn push_description_creates_new_release_when_pushed_severity_is_first_of_higher() {
+        let changelog_str = r"# Changelog
+
+## 0.1.1-git
+
+### Minor
+
+- A change
+
+## 0.1.0
+
+<!-- Increment to skip CHANGELOG.md test: 0 -->
+";
+
+        let mut changelog =
+            Changelog::parse("", changelog_str).expect("Failed to parse changelog.");
+
+        changelog.push_description(Severity::Major, "asdf").expect("Failed to push description.");
+
+        assert_eq!(changelog.get_current_release().version, Version::parse("0.2.0-git").unwrap());
```

Comment:
We broke the invariant that we have only one pre-release per changelog. It's not enough to check the latest release.